src/praisonai/praisonai/bots/_session.py

```
@@ -661,6 +668,21 @@ def bridged_stream_callback(event):
+                    # (text, presentation). Agents/hooks may return a plain str
+                    # (unchanged), a MessagePresentation, or an AgentReply. The
+                    # portable presentation is captured per-user so channel
+                    # adapters can render interactive UI; the text is returned as
+                    # before so the str contract and text fallback are preserved.
+                    try:
+                        from praisonaiagents.bots.agent_reply import extract_presentation
+                        text, presentation = extract_presentation(response)
+                        if presentation is not None:
+                            self._last_presentation[self._storage_key(user_id)] = presentation
+                            response = text
+                    except Exception as e:  # pragma: no cover — defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)